eventuals/task.h

```diff
@@ -544,83 +626,29 @@ class TaskFromToWith {
     return future.get();
   }
 
- private:
-  _TaskFromToWith::Composable<From_, To_, Args_...> e_;
-
-  // NOTE: if 'Task::Start()' is invoked then 'Task' becomes not just
-  // a composable but also a continuation which has a terminal made up
-  // of the callbacks passed to 'Task::Start()'.
-  using K_ = decltype(Build(
-      std::move(e_)
-      | Terminal()
-            .start(std::declval<
-                   std::conditional_t<
-                       std::is_void_v<To_>,
-                       Callback<>&&,
-                       Callback<To_>&&>>())
-            .fail(std::declval<Callback<std::exception_ptr>&&>())
-            .stop(std::declval<Callback<>&&>())));
-
-  // NOTE: we store 'k_' as the _last_ member so it will be
-  // destructed _first_ and thus we won't have any use-after-delete
-  // issues during destruction of 'k_' if it holds any references or
-  // pointers to any (or within any) of the above members.
-  std::optional<K_> k_;
-};
-
-////////////////////////////////////////////////////////////////////////
-
-struct Task final {
-  template <typename From_>
-  struct From final : public TaskFromToWith<From_, void> {
-    template <typename To_>
-    struct To final : public TaskFromToWith<From_, To_> {
-      template <typename... Args_>
-      using With = TaskFromToWith<From_, To_, Args_...>;
-
-      To(MonostateIfVoidOr<To_> value)
-        : TaskFromToWith<From_, To_>(std::move(value)) {}
-
-      template <typename F>
-      To(F f)
-        : TaskFromToWith<From_, To_>(std::move(f)) {}
-
-      To(To&& that) = default;
-
-      ~To() override = default;
-    };
-
-    template <typename... Args_>
-    using With = TaskFromToWith<From_, void, Args_...>;
-
-    template <typename F>
-    From(F f)
-      : TaskFromToWith<From_, void>(std::move(f)) {}
-
-    From(From&& that) = default;
-
-    ~From() override = default;
-  };
-
-  template <typename To_>
-  using Of = From<void>::To<To_>;
-
-  template <typename... Args_>
-  using With = From<void>::To<void>::With<Args_...>;
-
   // Helpers for synchronous tasks.
+
   template <typename Value>
   static auto Success(Value value) {
-    return TaskFromToWith<void, Value>(std::move(value));
+    return TaskFromToWith<
+        void,
+        Value,
+        std::tuple<>>(std::move(value));
   }
 
-  template <typename Value>
+  template <typename Value, typename... Errors>
```

Comment:
Success never has errors!
```suggestion
  template <typename Value>
```